tests/unit/test_arithmetic_expression.cpp

```diff
@@ -1411,3 +1411,20 @@ TEST_F(ArithmeticTest, ParamTest) {
 	for(int i = 0; i < 3; i++) SIArray_Append(&array, SI_LongVal(i));
 	ASSERT_EQ(0, SIValue_Compare(array, arExp->operand.constant, NULL));
 }
+
+TEST_F(ArithmeticTest, CoalesceTest) {
+	const char *query;
+	AR_ExpNode *arExp;
+
+	query = "RETURN coalesce(1)";
+	arExp = _exp_from_query(query);
+	ASSERT_EQ(AR_EXP_OPERAND, arExp->type);
+	ASSERT_EQ(AR_EXP_CONSTANT, arExp->operand.type);
+	ASSERT_EQ(0, SIValue_Compare(SI_LongVal(1), arExp->operand.constant, NULL));
```

Comment:
This tests reducibility, reducing the function to a constant.

Reply:
added comments